Audiobook Player/Player/Containers/PlayerControlsViewController.swift

```diff
@@ -8,69 +8,225 @@
 
 import UIKit
 
-class PlayerControlsViewController: PlayerContainerViewController {
+class PlayerControlsViewController: PlayerContainerViewController, UIGestureRecognizerDelegate {
+    @IBOutlet private weak var artworkView: UIView!
     @IBOutlet private weak var artwork: UIImageView!
     @IBOutlet private weak var playPauseButton: UIButton!
-    @IBOutlet private weak var rewindButton: UIButton!
-    @IBOutlet private weak var forwardButton: UIButton!
+    @IBOutlet private weak var rewindIcon: PlayerForwardIconView!
+    @IBOutlet private weak var forwardIcon: PlayerRewindIconView!
+    @IBOutlet private weak var artworkHeight: NSLayoutConstraint!
+    @IBOutlet private weak var artworkHorizontal: NSLayoutConstraint!
+    @IBOutlet private weak var forwardIconHorizontal: NSLayoutConstraint!
+    @IBOutlet private weak var rewindIconHorizontal: NSLayoutConstraint!
+
+    private let playImage = UIImage(named: "playButton")
+    private let pauseImage = UIImage(named: "pauseButton")
+    private var pan: UIPanGestureRecognizer!
+    private var originalHeight: CGFloat!
+    private let jumpIconAlpha: CGFloat = 0.15
+    private var triggeredPanAction: Bool = false
+
+    private var isPlaying: Bool = false {
+        didSet {
+            self.playPauseButton.alpha = 1.0
+            self.playPauseButton.setImage(self.isPlaying ? self.pauseImage : self.playImage, for: UIControlState())
+
+            self.view.layoutIfNeeded()
+            self.artworkHeight.constant = self.isPlaying ? self.originalHeight : self.originalHeight * 255/325
```

Comment:
what is `255` or `325` about?

Reply:
Magic value extracted from the design to determine how big the artwork should be for the scaled state. I should add a comment to explain or move the values to variables explaining what they represent (325 being the width of the artwork in the design in its default state and 255 being the scaled artwork)